### ISSUE TITLE
Error on selectionMarker.destroy

### ISSUE THREAD

**Issue by glittle** [highcharts/highcharts#5431]
Sample, showing this problem: http://jsfiddle.net/glittle/cg92p6o9/  (When running, an exception is thrown when zooming. See in Developer Tools...)

I have a chart where I'm doing some custom zooming:
```
chart: {
        renderTo: 'chart',
        zoomType: 'x',
        events: {
          selection: function (ev) {
            ev.preventDefault();
            customZoom();
          }
        },
```

Sometimes in the `customZoom()` method, I need to entirely recreate the entire chart. That works quite well... except for the code below (in highcharts.src.js):
```
if (this.selectionMarker) {
...
       if (runZoom) {
           fireEvent(chart, 'selection', selectionData, function (args) {
               chart.zoom(extend(args, hasPinched ? { animation: false } : null));
           });
       }

   }
   this.selectionMarker = this.selectionMarker.destroy();
// Reset all
if (chart) { // it may be destroyed on mouse up - #877
...
```
When the user selects a range to zoom in on, this code is run, which fires the `selection` event. My code is running inside of the selection event and ends up using `chart.destroy()` on the original chart and making a new one. When the event is done, the next line (above) runs: `this.selectionMarker.destroy()`.  However, by this time the selectionMarker is gone, since the previous chart was destroyed.

The very next line checks to see if the chart still exists. Something like that needs to be done here as well.  Unfortunately, an empty `chart` object does exist 
![image](https://cloud.githubusercontent.com/assets/85186/16319632/38d31ea4-3951-11e6-9c7f-2b5ac44c7283.png)

I would suggest checking to ensure the object still exists:
`this.selectionMarker = this.selectionMarker ? this.selectionMarker.destroy() : null;`

And `if(chart)` would not work in this case either, since it does exist as a `Highcharts.Chart` object, but is entirely empty.

**Comment by glittle** [comment]
#877

**Comment by jon-a-nygaard** [comment]
Already fixed in bugfix branch but not picked into the master branch. Commit with fix is 0d8be84.

This issue is a duplicate of #3535, therefore I am closing this issue.
